### PR TITLE
Feat Added Filters

### PR DESCRIPTION
**Pull request by AmishMishra11**
Added Filters to the product listing page
Features:

- Select Category from the home page
- select category from the product listing page
- choose the rating of the product
- choose the price range of the product
-  sort products by price

https://user-images.githubusercontent.com/76205249/159115474-02a19eeb-9387-47b7-9189-7a125c0696fa.mp4

### CODE REVIEW

**Review by spraveenofficial — APPROVED** [review]

File: src/utilities/categoryFunction.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const categoryFunction = (arrOfData, men, women, kid, toy) => {
+  const filteredData = [];
+
+  if (men) {
```
Comment (tanishk15sharma) [inline src/utilities/categoryFunction.js:4]: avoid hardcoding if possible.

File: src/utilities/rangeFunction.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const rangeFunction = (arrOfData, range) => {
+  if (range === 3)
```
Comment (tanishk15sharma) [inline src/utilities/rangeFunction.js:2]: Use dynamic code if possible writing `if` `else` again and again is not a good practice .

File: src/utilities/sortingFunction.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const sortingFunction = (arrOfData, sorting) => {
```
Comment (tanishk15sharma) [inline src/utilities/sortingFunction.js:1]: This function is appropriate .

**Review by tanishk15sharma — COMMENTED** [review]
Suggested some new measures to write good code , other things looks fine to me .

File: src/utilities/categoryFunction.js (hunk 1 above)
Comment (AmishMishra11) [reply src/utilities/categoryFunction.js:4]: resolved via voice call

File: src/utilities/rangeFunction.js (hunk 2 above)
Comment (AmishMishra11) [reply src/utilities/rangeFunction.js:2]: resolved via voice call

File: src/utilities/sortingFunction.js (hunk 3 above)
Comment (AmishMishra11) [reply src/utilities/sortingFunction.js:1]: Thanks

File: src/utilities/categoryFunction.js (hunk 1 above)
Comment (tanishk15sharma) [reply src/utilities/categoryFunction.js:4]: yes ,

**Pull request merged**